### DISCUSSION TITLE
Issues after updating Bonsai

### DISCUSSION THREAD

**Discussion by benefron** [bonsai-rx/bonsai#1033] [Q&A]
Hello,
I updated all the packages for Bonsai (All sources)
When I am opening Bonsai with one of my projects or even create a new project I get a big red X all over the workflow and a warning message.
I am adding a screen capture and the full text of the warning.
I a updated the packages because of an issue with python transform node that worked before and suddenly was not recognized anymore.
Would appreciate some help ASAP since I can not run my experiments in the current situation  
[Bonsai_crash_120922.txt](https://github.com/bonsai-rx/bonsai/files/9545938/Bonsai_crash_120922.txt)
![1662969413578](https://user-images.githubusercontent.com/29516660/189601954-9276bc59-c0b3-42ce-bc73-6d7bd5d5b7f5.jpg)

**Comment by firsttimegituser** [comment]
same problem after updating bonsai 2.6.3. tried to get earlier versions 2.4.1 and 2.5 for Neurophotometrics compatibility but I get an error that Bonsai version is not found in the launching process

**Reply by bruno-f-cruz** [reply]
Can you attach your `Bonsai.config` file? It will be located next to the Bonsai executable in its default installation folder. Meanwhile, if you would like to have Bonsai up and running asap delete the Bonsai folder and reinstall from scratch.

**Reply by benefron** [reply]
[Bonsai.txt](https://github.com/bonsai-rx/bonsai/files/9547261/Bonsai.txt)
I attached it. converted it to txt cause github doesn't accept .config files

**Reply by firsttimegituser** [reply]
This error comes up only after updating all the packages. I could still see the workflow fine after installing all packages from scratch and clean installation of bonsai 2.6.3. But as soon as I update packages this red cross comes up

![image](https://user-images.githubusercontent.com/113070373/189637468-7899d822-afe7-422a-951c-6ece19f6a581.png)

Here is the error message and screen shot after installing pointgrey2 as it seemed the error was referring to, with no removal of error

![image](https://user-images.githubusercontent.com/113070373/189637979-079134b2-530c-4590-a131-07aa6aa4bb8b.png)


See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.TypeLoadException: Could not load type 'SvgNet.SvgGdi.GdiGraphics' from assembly 'SVG, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at Bonsai.Editor.GraphView.GraphViewControl.canvas_Paint(Object sender, PaintEventArgs e)
   at System.Windows.Forms.Control.OnPaint(PaintEventArgs e)
   at System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   at System.Windows.Forms.Control.WmPaint(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4515.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Bonsai
    Assembly Version: 2.6.3.0
    Win32 Version: 2.6.3.3552
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Bonsai.EXE
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Bonsai.Editor
    Assembly Version: 2.6.3.0
    Win32 Version: 2.6.3.3552
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Editor.2.6.3/lib/net462/Bonsai.Editor.dll
----------------------------------------
Bonsai.Core
    Assembly Version: 2.6.3.0
    Win32 Version: 2.6.3.3723
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Core.2.6.3/lib/net462/Bonsai.Core.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4550.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4390.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Reactive.Linq
    Assembly Version: 2.2.5.0
    Win32 Version: 2.2.5.40722
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Rx-Linq.2.2.5/lib/net45/System.Reactive.Linq.dll
----------------------------------------
System.Reactive.Interfaces
    Assembly Version: 2.2.5.0
    Win32 Version: 2.2.5.40722
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Rx-Interfaces.2.2.5/lib/net45/System.Reactive.Interfaces.dll
----------------------------------------
System.Reactive.Core
    Assembly Version: 2.2.5.0
    Win32 Version: 2.2.5.40722
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Rx-Core.2.2.5/lib/net45/System.Reactive.Core.dll
----------------------------------------
System.Reactive.PlatformServices
    Assembly Version: 2.2.5.0
    Win32 Version: 2.2.5.40722
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Rx-PlatformServices.2.2.5/lib/net45/System.Reactive.PlatformServices.dll
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
Bonsai.Design
    Assembly Version: 2.6.1.0
    Win32 Version: 2.6.1.3385
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Design.2.6.1/lib/net462/Bonsai.Design.dll
----------------------------------------
System.IO.Compression
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.IO.Compression/v4.0_4.0.0.0__b77a5c561934e089/System.IO.Compression.dll
----------------------------------------
System.Net.Http
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Bonsai.PointGrey2
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.PointGrey2.1.0.0/lib/net45/Bonsai.PointGrey2.dll
----------------------------------------
@DynamicBonsai.PointGrey2
    Assembly Version: 1.0.0.0
    Win32 Version: 2.6.3.3723
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Core.2.6.3/lib/net462/Bonsai.Core.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
Bonsai.Vision
    Assembly Version: 2.6.1.0
    Win32 Version: 2.6.1.3552
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Vision.2.6.1/lib/net462/Bonsai.Vision.dll
----------------------------------------
Bonsai.System
    Assembly Version: 2.6.0.0
    Win32 Version: 2.6.0.3290
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.System.2.6.0/lib/net462/Bonsai.System.dll
----------------------------------------
OpenCV.Net
    Assembly Version: 3.4.1.0
    Win32 Version: 3.4.1.0
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/OpenCV.Net.3.4.1/lib/net462/OpenCV.Net.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4536.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
Bonsai.Scripting
    Assembly Version: 2.6.1.0
    Win32 Version: 2.6.1.3552
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Scripting.2.6.1/lib/net462/Bonsai.Scripting.dll
----------------------------------------
Bonsai.Harp
    Assembly Version: 3.3.0.0
    Win32 Version: 3.3.0.0
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Harp.3.3.0/lib/net462/Bonsai.Harp.dll
----------------------------------------
Bonsai.Harp.CF
    Assembly Version: 1.6.0.0
    Win32 Version: 1.6.0.0
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Harp.CF.1.6.0/lib/net45/Bonsai.Harp.CF.dll
----------------------------------------
Bonsai.Dsp
    Assembly Version: 2.6.1.0
    Win32 Version: 2.6.1.3552
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Dsp.2.6.1/lib/net462/Bonsai.Dsp.dll
----------------------------------------
Bonsai.Arduino
    Assembly Version: 2.5.1.0
    Win32 Version: 2.5.1.3385
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/Bonsai.Arduino.2.5.1/lib/net462/Bonsai.Arduino.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
SVG
    Assembly Version: 3.0.0.0
    Win32 Version: 3.0.0.0
    CodeBase: file:///C:/Users/Scoobie/AppData/Local/Bonsai/Packages/SvgNet.3.0.0/lib/net461/SVG.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Reply by firsttimegituser** [reply]
Here is my config files. I searched and came up with two Bonsai.config . One is in Bonsai folder and the other is in a subfolder within Nuget. Is this the problem?

![image](https://user-images.githubusercontent.com/113070373/189639497-1f3b7fc3-4251-44e4-a091-7bfe91fe497e.png)

[firstBonsaiconfigtext.txt](https://github.com/bonsai-rx/bonsai/files/9547419/firstBonsaiconfigtext.txt)

[secondbonsaiconfiginnuget.txt](https://github.com/bonsai-rx/bonsai/files/9547420/secondbonsaiconfiginnuget.txt)

**Reply by glopesdev** [reply]
@firsttimegituser can you share also your NuGet.config file? it should be inside the NuGet folder in the install dir. Alternatively, can you go to Tools > Manage Packages, then click on the little cogwheel to the right of "Package Source" and share a screenshot of which sources you have configured?

![image](https://user-images.githubusercontent.com/5315880/189641081-9eea5604-4486-45a1-929e-d37b06fbe120.png)

You might be downloading updates from nuget.org and not only from the official Bonsai package feed, and it looks like you are updating SvgNet to a later version than what is supported (3.0.0).

**Reply by firsttimegituser** [reply]
![image](https://user-images.githubusercontent.com/113070373/189642438-7491b83e-b4e3-45c7-a405-d743f876b669.png)

**Reply by firsttimegituser** [reply]
![image](https://user-images.githubusercontent.com/113070373/189642861-8379c691-eba2-40cb-ac1c-e411170c0919.png)

I see two NuGet.config files in my computer

**Reply by firsttimegituser** [reply]
[FirstNuGetonList.txt](https://github.com/bonsai-rx/bonsai/files/9547537/FirstNuGetonList.txt)
[SecondNuGetonLIst.txt](https://github.com/bonsai-rx/bonsai/files/9547539/SecondNuGetonLIst.txt)

**Reply by firsttimegituser** [reply]
> @firsttimegituser can you share also your NuGet.config file? it should be inside the NuGet folder in the install dir. Alternatively, can you go to Tools > Manage Packages, then click on the little cogwheel to the right of "Package Source" and share a screenshot of which sources you have configured?
> 
> ![image](https://user-images.githubusercontent.com/5315880/189641081-9eea5604-4486-45a1-929e-d37b06fbe120.png)
> 
> You might be downloading updates from nuget.org and not only from the official Bonsai package feed, and it looks like you are updating SvgNet to a later version than what is supported (3.0.0).

How do I fix this problem? Thanks

**Reply by benefron** [reply]
Do you need me to also upload the NuGet.config file?

**Reply by glopesdev** [reply]
@firsttimegituser can you try unchecking nuget.org from your settings list, and then installing again from scratch (i.e. make sure to delete the entire folder in the AppData) and updating?

**Reply by firsttimegituser** [reply]
> @firsttimegituser can you try unchecking nuget.org from your settings list, and then installing again from scratch (i.e. make sure to delete the entire folder in the AppData) and updating?

Sure. Can I confirm what is the proper way to install from scratch? 

I had previously use the Remove App function on Windows to uninstall Bonsai with one click. 

What folders are we supposed to delete manually (I know you mentioned AppData. Anything else?)

**Reply by glopesdev** [reply]
The uninstall function should now indeed remove all relevant folders. However if the installed version was too old, this feature might not have been in place, so it's best to confirm. Just deleting the Bonsai folder in AppData\Local\Bonsai should be enough (don't delete the entire AppData folder 🙂 ).

**Comment by firsttimegituser** [comment]
So I reinstalled Bonsai 2.6.3 and then updated only things not from nuget or gallery. there is no updates available anymore and i can see my bonsai workflow. however, there is another error in the bottom of the bar. it regards something with point grey. I cannot find the FlyCapture node in the list of nodes despite having installed PointGrey2 .

![image](https://user-images.githubusercontent.com/113070373/189646199-705e33ce-8b23-4230-8c02-566124ad8cef.png)


Any idea what could be wrong? And am I installing appropriate packages this time?

\
![image](https://user-images.githubusercontent.com/113070373/189646255-5eb4e432-f888-4c74-bb94-5eb06089501d.png)

**Comment by benefron** [comment]
@glopesdev Hi can you please check my bionsai.config file I uploaded?
Should I also just reinstall latest version of Bonsai or is there anything else to do?

**Reply by glopesdev** [reply]
@benefron yes, this seems to be the same issue as @firsttimegituser so if you reinstall Bonsai and make sure to disable the nuget.org folder when you update it should work well.
